src/mlpack/core/cv/cv_base.hpp

```diff
@@ -0,0 +1,329 @@
+/**
+ * @file cv_base.hpp
+ * @author Kirill Mishchenko
+ *
+ * A base class for cross-validation.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_CV_BASE_HPP
+#define MLPACK_CORE_CV_CV_BASE_HPP
+
+#include <mlpack/core/cv/meta_info_extractor.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/*
+ * A base class for cross-validation. It serves to handle basic non-data
+ * constructor parameters of a machine learning algorithm (like datasetInfo or
+ * numClasses) and to assert that the machine learning algorithm and data
+ * satisfy certain conditions.
+ *
+ * @tparam MLAlgorithm A machine learning algorithm.
+ * @tparam MatType The type of data.
+ * @tparam PredictionsType The type of predictions (labels/responses).
+ * @tparam WeightsType The type of weights. It supposed to be void* when weights
+ *     are not supported.
+ */
+template<typename MLAlgorithm,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+class CVBase
```

Comment:
This is just a clarification question, why do we need a `CVBase` class?  We have `SimpleCV` that inherits from it, but I don't see anything else yet.  I don't mean to say that a `CVBase` class isn't a good idea, I just don't have enough information to know yet, so I thought I would ask to find out. :)